spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AuditRecordController.java

```diff
@@ -86,7 +89,38 @@ public AuditRecordController(AuditRecordService auditRecordService) {
 			@RequestParam(required = false) AuditOperationType[] operations,
 			PagedResourcesAssembler<AuditRecord> assembler) {
 		Page<AuditRecord> auditRecords = this.auditRecordService
-			.findAuditRecordByAuditOperationTypeAndAuditActionType(pageable, actions, operations);
+				.findAuditRecordByAuditOperationTypeAndAuditActionType(pageable, actions, operations);
+		return assembler.toResource(auditRecords, new Assembler(auditRecords));
+	}
+
+	/**
+	 * Return a page-able list of {@link AuditRecordResource}s depending on the date
+	 * parameters. The parser uses {@code DateTimeFormatter.ISO_DATE_TIME} to parse the String
+	 * parameters.
+	 *
+	 * If fomDate is null, the endpoint will return the {@link AuditRecord}s from the first
+	 * record until the given date. If toDate is null, the endpoint will return the
+	 * {@link AuditRecord}s from the given date to the last record. If both of the parameters
+	 * are not null the endpoint will return the {@link AuditRecord}s between the given range.
+	 * If both of the parameters are null it will return all of the inserted
+	 * {@link AuditRecord}s.
+	 *
+	 * @param pageable Pagination information
+	 * @param assembler assembler for {@link AuditRecord}
+	 * @param fromDate Optional. The from date in ISO_DATE_TIME format. eg.:
+	 *     2019-01-01T19:30:00.000-01:00
+	 * @param toDate Optional. The to date in ISO_DATE_TIME format. eg.:
+	 *     2019-01-04T19:30:00.000-01:00
+	 * @return list of audit records
+	 */
+	@RequestMapping(value = "query", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public PagedResources<AuditRecordResource> listByDate(Pageable pageable,
+			@RequestParam(required = false) String fromDate,
+			@RequestParam(required = false) String toDate,
+			PagedResourcesAssembler<AuditRecord> assembler) {
+		Page<AuditRecord> auditRecords = this.auditRecordService
```

Comment:
@sylank Thanks for your PR!!! Could you possibly try to add the `fromDate` and `toDate` request parameters to the existing Controller method? I think it would be cool to not only search by dates but also search by dates and `AuditActionType` and/or `AuditOperationType`.

I know this will likely mean a more complicated DAO and possibly looking at the criteria api for building dynamic persistence queries but I think that will lead to a cleaner more powerful rest endpoint overall.

Reply:
Thank you for the review, I will modify the controller class